src/core/include/openvino/core/any.hpp

```diff
@@ -621,8 +654,74 @@ class OPENVINO_API Any {
      * @return casted object
      */
     template <class T>
-    const typename std::enable_if<!std::is_convertible<T, std::shared_ptr<RuntimeAttribute>>::value, T>::type& as()
-        const& {
+    const typename std::enable_if<!std::is_convertible<T, std::shared_ptr<RuntimeAttribute>>::value &&
+                                      !std::is_same<T, std::string>::value && std::is_default_constructible<T>::value,
+                                  T>::type&
+    as() const& {
+        impl_check();
+        if (_impl->type_info() == typeid(decay_t<T>)) {
+            return *static_cast<const decay_t<T>*>(_impl->addressof());
+        } else if (_impl->type_info() == typeid(std::string)) {
+            _temp_impl = std::make_shared<Impl<decay_t<T>>>();
+            std::stringstream strm{as<std::string>()};
+            _temp_impl->read(strm);
```

Comment:
agreed offline to make more strict contract here and convert only string which represent numbers (inference / decimal) to actual numbers, other types have fallback to current `read`

Reply:
and let's add tests for it :)

Reply:
done